mne/io/base.py

```diff
@@ -2009,3 +2011,109 @@ def _check_update_montage(info, montage):
+        the raw file.
+    misc : list or tuple of str
+        Names of channels or list of indices that should be designated
+        MISC channels. Values should correspond to channels listed within
+        the raw file.
+    reference : bool | str
+        If True, add average EEG reference projector (if it's not already
+        present).
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    raw : Instance of Raw
+        A Raw object.
+    """
+
+    if isinstance(input_fname, str):
+        fname, ext = os.path.splitext(input_fname)
+    elif isinstance(input_fname, list):
+        for f in input_fname:
+            fname, ext = os.path.splitext(input_fname)
+            err = "File must be '.fif', not %s." % ext
+            assert_equal(ext, '.fif', err)
+
+        if eog is not None:
+            eog_map = dict()
+            for s in eog:
+                eog_map[s] = 'eog'
+        if misc is not None:
+            misc_map = dict()
+            for s in misc:
+                misc_map[s] = 'misc'
+
+    # (MEG) KIT systems
+    if ext in ['.con', '.sqd']:
+        raw = read_raw_kit(input_fname, mrk=montage, eog=eog, misc=misc,
+                           preload=preload, verbose=verbose)
+        if eog is not None:
+            rename_sensor(raw.info, eog_map)
+        if misc is not None:
+            rename_sensor(raw.info, misc_map)
+    # (EEG) EDF and Biosemi systems
+    elif ext == '.edf':
+        raw = read_raw_edf(input_fname, montage=montage, eog=eog, misc=eog,
+                           preload=preload, verbose=verbose)
+    # (EEG) EGI systems
+    elif ext == '.raw':
+        raw = read_raw_egi(input_fname, montage=montage, eog=eog, misc=misc,
+                           preload=preload, verbose=verbose)
+    # (MEG) Neuromag or converted-to-fif systems
+    elif ext == '.fif':
+        raw = read_raw_fif(input_fname, add_eeg_ref=reference, eog=eog,
+                           misc=misc, preload=preload, verbose=verbose)
+        if eog is not None:
+            rename_sensor(raw.info, eog_map)
+        if misc is not None:
+            rename_sensor(raw.info, misc_map)
+    # (MEG) BTi systems
+    elif ext == '.pdf':
+        raw = read_raw_bti(input_fname, eog=eog, misc=['E31'], preload=preload,
+                           verbose=verbose)
+        if eog is not None:
+            rename_sensor(raw.info, eog_map)
+        if misc is not None:
+            rename_sensor(raw.info, misc_map)
```

Comment:
does BTi data have an extension? I was guessing at one. I looked through the module and it didn't seem like the test data had one. if not, I guess since it is the only one without an extension, if `op.splitext` doesn't return anything, we could assume that it is this format.

yeah, I was trying to figure out the best way that case with BTi with those parameters. so for a params arg, you propose something like a dict for the add'l ones?